packages/cozy-client/src/hooks.js

```diff
@@ -1,7 +1,49 @@
-import { useContext } from 'react'
+import { useEffect, useContext } from 'react'
+import { useSelector } from 'react-redux'
 import clientContext from './context'
 
 export const useClient = () => {
   const { client } = useContext(clientContext)
   return client
 }
+
+const resolveQueryDefinition = query => {
+  const value = query.query || query.definition
+  return typeof value === 'function' ? value() : value
+}
+
+export const useQuery = ({ query }) => {
+  if (process.env.NODE_ENV !== 'prod' && !useSelector) {
+    throw new Error(
+      'You must use react-redux > 7.1.0 to use useQuery (uses useSelector) under the hood'
+    )
```

Comment:
Since react-redux is a cozy-client's dependency, if we update it, I think handling this case becomes useless, no?